Deprecation warnings fails PR build:
```[WARNING] COMPILATION WARNING : 
[INFO] -------------------------------------------------------------
[WARNING] /source/vespa-http-client/src/main/java/com/yahoo/vespa/http/client/runner/CommandLineArguments.java:[211,91] ALLOW_ALL_HOSTNAME_VERIFIER in org.apache.http.conn.ssl.SSLConnectionSocketFactory has been deprecated
[INFO] 1 warning
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /source/vespa-http-client/src/main/java/com/yahoo/vespa/http/client/runner/CommandLineArguments.java: warnings found and -Werror specified
[INFO] 1 error
[INFO] -------------------------------------------------------------
```

--- a/vespa-http-client/src/main/java/com/yahoo/vespa/http/client/runner/CommandLineArguments.java
+++ b/vespa-http-client/src/main/java/com/yahoo/vespa/http/client/runner/CommandLineArguments.java
@@ -11,6 +11,7 @@ import io.airlift.command.Command;
 import io.airlift.command.HelpOption;
 import io.airlift.command.Option;
 import io.airlift.command.SingleCommand;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
 import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
 
 import javax.inject.Inject;
@@ -208,7 +209,7 @@ public class CommandLineArguments {
                 )
                 .setConnectionParams(
                         new ConnectionParams.Builder()
-                                .setHostnameVerifier(insecure ? SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER :
+                                .setHostnameVerifier(insecure ? NoopHostnameVerifier.INSTANCE :
                                         SSLConnectionSocketFactory.getDefaultHostnameVerifier())
                                 .setNumPersistentConnectionsPerEndpoint(16)
                                 .setEnableV3Protocol(! enableV2Protocol)

